autopilot/contracts.go

```diff
@@ -1,16 +1,34 @@
 package autopilot
 
 import (
+	"time"
+
+	"gitlab.com/NebulousLabs/errors"
 	"go.sia.tech/renterd/internal/consensus"
+	rhpv2 "go.sia.tech/renterd/rhp/v2"
 	"go.sia.tech/renterd/worker"
 	"go.sia.tech/siad/types"
 	"golang.org/x/crypto/blake2b"
 )
 
+const contractSetName = "autopilot"
+
+func logErr(err error) {} // TODO
+
 func (ap *Autopilot) currentHeight() uint64 {
 	return ap.store.Tip().Height
 }
 
+func (ap *Autopilot) currentPeriod() uint64 {
+	c := ap.store.Config()
+	h := ap.store.Tip().Height
+	return h + c.Contracts.Period
+}
+
+// TODO: deriving the renter key from the host key using the master key only
+// works if we persist a hash of the renter's master key in the database and
+// compare it on startup, otherwise there's no way of knowing the derived key is
+// usuable
 func (ap *Autopilot) deriveRenterKey(hostKey consensus.PublicKey) consensus.PrivateKey {
```

Comment:
I feel like we need more explanation here. What is this key used for? Why is it derived from the host's? So many questions.

Reply:
this is the contract revision key. We hash it with the host's in order to get a unique key for each host, without having to store each key explicitly. (Now that I think about it, though, maybe we should instead generate a random salt and store it in the bus? I don't think we want to be limited to one key per host.)

Reply:
added TODO